### ISSUE TITLE
PipelineML breaks on kedro-viz 9.2.0

### ISSUE THREAD

**Issue by Calychas** [Galileo-Galilei/kedro-mlflow#569]
## Description

PipelineML breaks on kedro-viz 9.2.0

## Context

The new kedro-viz 9.2.0 release heavily refactors graph generation for namespaced pipelines - https://github.com/kedro-org/kedro-viz/pull/1897. It now uses `only_nodes_with_namespace` and `__sub__` pipeline methods, but PipelineML doesn't have them implemented. 

## Steps to Reproduce

1. Have a PipelineML specified in kedro project (by using `pipeline_ml_factory`)
2. Run kedro viz

## Expected Result

Kedro viz should run, and pipeline should display training nodes

## Actual Result

Kedro viz cannot run at all

```
  File "repo/venv/lib/python3.10/site-packages/kedro_viz/data_access/repositories/modular_pipelines.py", line 61, in populate_tree
    sub_pipeline = pipeline.only_nodes_with_namespace(modular_pipeline_id)
  File "repo/venv/lib/python3.10/site-packages/kedro_mlflow/pipeline/pipeline_ml.py", line 168, in only_nodes_with_namespace
    raise NotImplementedError(MSG_NOT_IMPLEMENTED)
NotImplementedError: This method is not implemented because it does not make sense for 'PipelineML'. Manipulate directly the training pipeline and recreate the 'PipelineML' with 'pipeline_ml_factory' factory.

```

```
  File "repo/venv/lib/python3.10/site-packages/kedro_viz/data_access/repositories/modular_pipelines.py", line 62, in populate_tree
    rest_of_the_pipeline = pipeline - sub_pipeline
  File "repo/venv/lib/python3.10/site-packages/kedro_mlflow/pipeline/pipeline_ml.py", line 239, in __sub__
    raise NotImplementedError(MSG_NOT_IMPLEMENTED)
NotImplementedError: This method is not implemented because it does not make sense for 'PipelineML'. Manipulate directly the training pipeline and recreate the 'PipelineML' with 'pipeline_ml_factory' factory.

```

## Workaround

Quick and dirty implementation in PipelineML class:

```py
def only_nodes_with_namespace(
    self, node_namespace: str
) -> "Pipeline":  # pragma: no cover
    return self.training.only_nodes_with_namespace(node_namespace)
```

```py
def __sub__(self, other):  # pragma: no cover
      return self.training - other
```

Can't use `self._turn_pipeline_to_ml(...)` here as the pipelines are not consistent anymore

## Your Environment

* `kedro` and `kedro-mlflow` : 0.19.6 and 0.12.2
* Python version used (`python -V`): 3.10.11
* Operating system and version: M2 Sonoma 14.5

## Does the bug also happen with the last version on master?

Probably yes

**Comment by Calychas** [comment]
Quick fix avaliable here - https://github.com/Calychas/kedro-mlflow/tree/kedro-viz-9-2-0

**Comment by Galileo-Galilei** [comment]
Thank you very much, I'll have a look tonight or tomorrow

**Comment by Galileo-Galilei** [comment]
@Calychas @kacper-ki Ccan you try ``pip install git+https://github.com/Galileo-Galilei/kedro-mlflow.git569-fix-kedro-viz`` and tell me if it fixes your issue?

**Comment by cariveroco** [comment]
> @Calychas @kacper-ki Ccan you try `pip install git+https://github.com/Galileo-Galilei/kedro-mlflow.git@569-fix-kedro-viz` and tell me if it fixes your issue?

@Galileo-Galilei I'm also having this issue so I tried your fix, but I'm getting `TypeError: 'Logger' object is not callable`. So I edited the package script directly on my local and replaced  `self._logger(MSG_WARNING_KEDRO_VIZ)` in lines 181 and 252 of `kedro_mlflow/pipeline/pipeline_ml.py`  with `self._logger.warning(MSG_WARNING_KEDRO_VIZ)`. It worked and I have no issues with kedro-viz 9.2.0 now. I'm using Python 3.10.14 if that matters.
